lib/logstash/filters/drop.rb

```diff
@@ -26,6 +39,6 @@ def register
 
   public
   def filter(event)
-    event.cancel
+    event.cancel if rand < (@sample / 100.0)
```

Comment:
the `if @sample` requires the removal of `:default => ...` which is a good thing anyway :)